ams/server/src/main/java/com/netease/arctic/server/catalog/ExternalCatalog.java

```diff
@@ -54,6 +64,27 @@ protected ExternalCatalog(CatalogMeta metadata) {
     updateDatabaseFilter(metadata);
   }
 
+  public ArcticCatalog getArcticCatalog() {
+    FormatCatalog formatCatalog =
+        ((CommonUnifiedCatalog) unifiedCatalog)
+            .formatCatalogAsOrder(TableFormat.MIXED_HIVE, TableFormat.MIXED_ICEBERG)
+            .findFirst()
+            .get();
+    if (formatCatalog instanceof MixedCatalog) {
+      return ((MixedCatalog) formatCatalog).getCatalog();
+    }
+    return null;
+  }
+
+  public HMSClientPool getHMSClientPool() {
+    // external catalog also need show hive tables
+    ArcticCatalog arcticCatalog = getArcticCatalog();
+    if (arcticCatalog instanceof ArcticHiveCatalog) {
+      return ((ArcticHiveCatalog) arcticCatalog).getHMSClient();
+    }
+    return null;
+  }
+
```

Comment:
I don't think that we should get HMSClient in this way. 

HMSClient is constructed by TableMetastore and catalog properties, the ExternalCatalog object has those two fields. 


```
public HMSClientPool getHMSClientPool() {
    return new CachedHiveClientPool(tableMetastore, getMetadata());
}
```